### PR TITLE
`cncint` `cnclon` and `cncnum` fixes

### PR DESCRIPTION
**Pull request by paladine**
Adds an integration test with an updated `MBBSEMU.DLL` as well. Module uses `cnclon` to extract a value and writes it to btrieve, so also helps test the btrieve insertion code.

### CODE REVIEW

**Review by enusbaum — CHANGES_REQUESTED** [review]

File: MBBSEmu.Tests/Integration/BtrievePrint_Tests.cs [hunk 1]
```diff
@@ -37,5 +37,59 @@ public void DoEchoTestAndLogOff()
                 WaitUntil('.', "Have a nice day");
             });
         }
+
+        private static int mbbsemuHash(string s)
+        {
+            int hash = 7;
+            foreach (char c in s)
```
[the remark this answers is not in the document]
Comment (paladine) [reply MBBSEmu.Tests/Integration/BtrievePrint_Tests.cs:44]: That's a negative Ghost Rider

File: MBBSEmu/HostProcess/ExportedModules/ExportedModuleBase.cs [hunk 2]
```diff
@@ -896,39 +902,54 @@ protected bool ConsumeWhitespace(IEnumerator<char> input)
             return (builder.ToString(), false);
         }
 
+        protected class LeadingNumberFromStringResult
+        {
+            public bool Valid { get; set; }
+            public int Value { get; set; }
+            public String StringValue { get; set; }
```
[the remark this answers is not in the document]
Comment (paladine) [reply MBBSEmu/HostProcess/ExportedModules/ExportedModuleBase.cs:909]: Intermixing Java & C# is confusing

File: MBBSEmu/HostProcess/ExportedModules/ExportedModuleBase.cs [hunk 3]
```diff
@@ -896,39 +902,54 @@ protected bool ConsumeWhitespace(IEnumerator<char> input)
...
                 if (first && c == '+')
                     return CharacterAccepterResponse.SKIP;
-                if (char.IsDigit(c) || (first && c == '-'))
+                else if ((first && c == '-') || char.IsDigit(c))
                     return CharacterAccepterResponse.ACCEPT;
 
                 return CharacterAccepterResponse.ABORT;
             });
 
+            result.Valid = Int32.TryParse(result.StringValue, out var value);
```
[the remark this answers is not in the document]
Comment (paladine) [reply MBBSEmu/HostProcess/ExportedModules/ExportedModuleBase.cs:948]: not following on this one

[the remark this answers is not in the document]
Comment (enusbaum) [reply MBBSEmu/HostProcess/ExportedModules/ExportedModuleBase.cs:953]: int.TryParse()

**Review by enusbaum — APPROVED** [review]

**Pull request merged**